src/sql/src/plan/typeconv.rs

```diff
@@ -239,11 +239,13 @@ lazy_static! {
             (Timestamp, Date) => Assignment: CastTimestampToDate(func::CastTimestampToDate),
             (Timestamp, TimestampTz) => Implicit: CastTimestampToTimestampTz(func::CastTimestampToTimestampTz),
             (Timestamp, String) => Assignment: CastTimestampToString(func::CastTimestampToString),
+            (Timestamp, Time) => Assignment: CastTimestampToTime(func::CastTimestampToTime),
```

Comment:
I like to keep the cases roughly ordered by type category, so can you place these above the string casts?